graphql_compiler/compiler/compiler_frontend.py

```diff
@@ -525,6 +526,18 @@ def _compile_vertex_ast(schema, current_schema_type, ast,
     return basic_blocks
 
 
+def _validate_fold_has_outputs(fold_data, outputs):
+    # At least one output in the outputs list must point to the fold_data,
+    # or the scope corresponding to fold_data had no @outputs and is illegal.
+    for output in outputs.values():
+        if output['fold'] is fold_data:
+            return True
+    
+    raise GraphQLCompilationError(u'A @fold must contain at least one @output!'
```

Comment:
Nitpick: I'd expand the explanation here, in order to help out users. Maybe something like:
```
Each @fold scope must contain at least one field marked @output. Encountered a @fold with no outputs at query location: {}
```